src/FsCheck/Arbitrary.fs

```diff
@@ -438,13 +438,10 @@ module Arb =
         ///Actually, most of the values in range [0; 1) are NEVER generated.
         ///See "Generating uniform doubles in the unit interval" at http://xoshiro.di.unimi.it/ 
         static member private stdFloatGen =
-            let toFloat bytes =
-                BitConverter.ToUInt64(bytes, 0)
-                |> fun n -> (float (n >>> 11)) * (1.0 / float (1UL <<< 53))
-                |> BitConverter.GetBytes
-                |> fun b -> BitConverter.ToDouble(b, 0)
-            Gen.arrayOfLength 8 generate<byte>
-            |> Gen.map toFloat        
+            gen {
+                let! n = generate<DoNotSize<uint64>> |> Gen.map DoNotSize.Unwrap
+                return (float (n >>> 11)) * (1.0 / float (1UL <<< 53))
```

Comment:
I'm going to accept this as is, but looks like you don't need the `let!` here either (which will translate into `bind`) and can replace by another `Gen.map` call. This is also faster, also if you can get rid of the `gen {` computation expression altogether you'll save on another few thunks (like `Delay` etc).